apollo-router/src/lib.rs

```diff
@@ -101,4 +101,17 @@ pub mod _private {
     // For tests
     pub use crate::plugins::telemetry::Telemetry as TelemetryPlugin;
     pub use crate::router_factory::create_test_service_factory_from_yaml;
+
+    /// Retuns the `Debug` fomatting of two `Result<Schema, SchemaError>`,
+    /// from `parse_with_ast` and `parse_with_hir` respectively.
+    ///
+    /// The two strings are expected to be equal.
+    pub fn compare_schema_parsing(schema: &str) -> (String, String) {
```

Comment:
NIT: it's weird to call this function `compare_...` as it doesn't compare anything, it just parses the schema with 2 different ways

Reply:
Yeah that’s fair. It returns two strings for the caller to compare. Do you have a suggestion?

That’s left to the caller is so that differences can be printed with https://crates.io/crates/similar-asserts without making that crate a dependency of the router. https://github.com/apollographql/harness-vs-apollo-rs is a separate crate so it needs the router crate to provide some `pub` non-`#[cfg(test)]` API.

Anyway this is a semi-private API that we’ll remove eventually.